src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -327,13 +340,31 @@ ActStatus ActionHandlers::process_ula(SM::ControlBlock& cb)
 	ue_ctxt->setSubscriptionStatus(ula_msg.subscription_status);
 	ue_ctxt->setNetAccessMode(ula_msg.net_access_mode);
 	ue_ctxt->setAccessRestrictionData(ula_msg.access_restriction_data);
-  ue_ctxt->setSubscribedApn(Apn_name(ula_msg.selected_apn));
-  log_msg(LOG_DEBUG, "Selected APN %s \n",ula_msg.selected_apn.val);
+	
+    ue_ctxt->setSubscribedApn(Apn_name(ula_msg.selected_apn));
+    log_msg(LOG_DEBUG, "Selected APN %s \n", ula_msg.selected_apn.val);
 
+    for (int i = 0; i < ula_msg.supp_features_list.count; i++)
+    {
+        if (ula_msg.supp_features_list.supp_features[i].feature_list_id == 2)
+        {
+            ue_ctxt->setHssFeatList2(
+                    ula_msg.supp_features_list.supp_features[i]);
+            break;
+        }
+    }
+        
+	ue_ctxt->setDcnrCapable(MmeCommonUtils::isUeNRCapable(*ue_ctxt));
```

Comment:
Why this code ? If ue Is NR capable then DCNR capable ?

Reply:
Even if the UE advertises its capability, the network may/may not support the feature.
This flag indicates the result of the negotiation, that takes place to confirm whether the network supports the feature or not.